index.js

```diff
@@ -32,19 +35,23 @@ export default function pLimit(concurrency) {
 	};
 
 	const enqueue = (function_, resolve, arguments_) => {
-		queue.enqueue(
-			AsyncResource.bind(run.bind(undefined, function_, resolve, arguments_)),
+		// Queue `resolve` function (here `r` function) instead of `run` function
+		// to preserve asynchronous context.
+		new Promise(r => {
```

Comment:
`r` needs a proper variable name, not one character.

Reply:
However, if the `r' is made into a word, it becomes `resolve`, which is confusing with the `resolve` argument. This would be the same for `_resolve` and `resolve2`. I used `r` so you can tell the difference at a glance.

If you have a better name, please let me know.